packages/api-clients/open-api/m2mGatewayApi.yml

```
@@ -3535,6 +3535,87 @@ paths:
               $ref: "#/components/headers/RateLimitRemainingHeader"
             X-Rate-Limit-Interval:
               $ref: "#/components/headers/RateLimitIntervalHeader"
+    delete:
+      tags:
+        - purposes
+      summary: Delete Purpose
+      description: Delete Purpose. Only purposes with one version in Draft or WaitingForApproval state can be deleted
```

minor change for clarity, to avoid considering that the purpose may contain N versions, and only one of them is in the required states (feel free to improve the phrasing)
```suggestion
      description: Delete Purpose. Only purposes with one version can be deleted. The version must be in Draft or WaitingForApproval state
```